Automattic/simplenote-ios
# Feature/in app notifications

### Fix
This PR is to merge the In App Notifications feature into develop!!  The different components of this change have been tested along the way but here is a chance to test them all together again before moving into develop.  

In app notifications builds a tool that allows us to send simple messages to users about the activity in the app, such as copying a link or publishing a note.  Some of these messages have actions associated with them like undoing the previous change.  They look like this:

<img height="400px" src="https://cdn-std.droplr.net/files/acc_593859/bycPhS" />

### Test
Here are some things to test with these notifications:
1. copy a link (internal or public) you should see a notice
2. Copy a link from the options view, you should see a notice and the options view will be dismissed
3. Delete a note from the options view, a notice will be presented and the options view will be dismissed
4. delete a note from the notes list.  Tap on the undo button to make sure the note restores
5. In the options menu publish a note.  You should see two notices "Publishing..." and then "Publish Successful"  Make sure to tap the copy link button on the "Publish successful" notice
6. Unpublish a note and you should see two notices "unpublishing..." and "Unpublish Successful" with an undo action
7. Make sure that the notices stay aligned correctly if you rotate the device

### Review
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
> Added in app notifications to alert users of different activities in the app
